### ISSUE TITLE
Remove commit 4e47c8 (Remove Tibet from the map)

### ISSUE THREAD

**Issue by fengkaijia** [piwik/piwik#6006]
Please remove commit 4e47c8aba3bf146292f6e9d38aa0151c383e6c33.

This commit deeply hurt the Chinese Piwik community and should be immediately removed from Piwik.

**Comment by peanode** [comment]
Deeply agree +1

**Comment by jeehou** [comment]
Deeply agree +1

**Comment by JokerQyou** [comment]
I'm not saying which view of world is better or true, but this commit simply breaks the basic law here in China. Not only will it hurt Chinese piwik community, it might also prevent some users from using piwik, considering it's probably illegal.

**Comment by khlipeng** [comment]
Tibei and Taiwan should be a part of China

**Comment by mars333** [comment]
Deeply agree +1

**Comment by fordguo** [comment]
+1 ,it's  basic in china.

**Comment by joewulf** [comment]
fucking asshole

**Comment by JokerQyou** [comment]
@joewulf Please hold your anger and talk friendly, as offensive words will do nothing but break the discussion.

**Comment by joewulf** [comment]
You simply can't reason with the 'Pride and Prejudice'

**Comment by joewulf** [comment]
that piece of shit even remove Hainan from China in the detail map.

**Comment by sgiehl** [comment]
@joewulf please respect the [github terms of service](https://help.github.com/articles/github-terms-of-service/#3-conduct-restrictions)
If you would like to have any change in Piwik feel free to create a Pull Request.

**Comment by joewulf** [comment]
@sgiehl I'm sorry for my word, but not sorry for doing that.
NO PULL REQUEST CAN CHANGE A BIASED MIND.
That commit is just a personal decision.

**Comment by sgiehl** [comment]
> NO PULL REQUEST CAN CHANGE A BIASED MIND.

Guess that applies to you as well. 
Insult and hate are imho no acceptable ways of facing people with another opinion or mind.

**Comment by joewulf** [comment]
@sgiehl
Not gonna count on it to make any change.
Just speak out my voice and point out the simple fact: that commit is just a political act and respect no reality.
No need to remove the commit which lasts so many years, keep it to infinity.

**Comment by sunzhongmou** [comment]
This issue opened on Aug 15, 2014, and today is July 31, 2017.
It's so sad for Piwik performs unprofessional this way.

Refuse to using it anymore!!!

**Comment by sgiehl** [comment]
@sunzhongmou  this is an open source project. Feel free to create a pull reuqest.

**Comment by chaomao** [comment]
confused about this issue, does repo owner want to pretend deaf and ignore other people's opinion? That's funny because it remind me the kids behavior. :)
Open software should be away from politics and be purely tech, especially map software should respect country boundary.
If you are not sure about country boundary, you can follow google maps. :)

**Comment by chaomao** [comment]
@sgiehl do you think if I revert commit 4e47c8 and create pr, then pr will be accept? If you say yes, I will create one immediately.

**Comment by sgiehl** [comment]
@chaomao If it would be that easy I would have maybe done that already. Simply reverting the commit won't work and wouldn't be enough. Already tracked visits wouldn't change automatically. So guess there would be some adjustments in APIs needed to count them correctly or combine to the results with those from China.

**Comment by jijiechen** [comment]
@sgiehl  Good point. Then let's do it step by step.

1. Include Tibet and Taiwan in China on the map, which is a common sense. Tibet was never a country, it's a province of China; Taiwan is also a part of China, this is recognized by plenty of international laws.
2. Treat the visit and event data from these two provinces as part of China data. 

I have the same concern with @chaomao  how possible such PR could be accepted if we create them?

**Comment by fengkaijia** [comment]
@sgiehl @jijiechen @chaomao With @mattab's irreversible bias on this issue and a total lack of knowledge of OSS and world history, I don't think a simple revert pr will pass (for example, #6749).

As you said in #9719 the map generator isn't working anymore, plus @mattab's misdeeds in ecb3d2e, fd0632c, eb4699e etc., I won't waste hours on this bug plus the map generator unless @mattab give his solo promise that he will accept a pr that fix all these problems. If not, then I don't think we need talking to the wall anymore.

The free software movement was founded on the mutual respect b/w the contributor and the user. That respect includes honoring the existing facts. Tibet is a part of China. That's a fact recognized by every nation in the world, every map publisher in the world from the United Nations to Google Analytics. If an OSS contributor is arrogantly enough to ignore the facts and believe that he can influence the users with his prejudice, then he's a fool and he is harming the free software movement. Did Linus Torvalds make Linux incompatible with PC that has a Nvidia-card? He didn't because he respects the user, in return, the user respects him. And that's how the free software movement benefits the social progress and humanity.

And to @mattab:

![](http://s2.quickmeme.com/img/c4/c4cd24473dae2194eee8334bd8bfd11b5e73a57e34cae8fc4504fc82fdc774ab.jpg)

Please read the reference in this <https://en.wikipedia.org/wiki/Serfdom_in_Tibet_controversy> before copying your comment from #6749.

**Comment by hejiaji** [comment]
@mattab I really think you should learn more of histories before you post any comments

**Comment by jiaopianjun** [comment]
666666666666666666666666666

**Comment by jiaopianjun** [comment]
I really think you should learn more of histories before you post any comments

**Comment by zqinmiao** [comment]
Tibet is part of China!

**Comment by myloft** [comment]
Tibet is part of China!

**Comment by YuanYe24** [comment]
Tibet is part of China!

**Comment by xgz123** [comment]
诸位，我们不用他抵制他不行吗，自己开发一个比他好的不行吗？它能accept这个提交已经说明问题了。
let's boycott 'piwik'.

**Comment by justjavac** [comment]
Tibet is part of China!

**Comment by phodal** [comment]
Tibet is part of China!

**Comment by xiasun** [comment]
I wonder when it comes to different points of view, based on which knowledge are you going to decide whether accept it or not? 
Seriously this is important for an open source software developer.

**Comment by phodal** [comment]
@sgiehl, do you also mean the Germany should become GDR + FDR? In the 1959, there was two Germanys? 

Would you like in other version, you have two Germany??

**Comment by mirsking** [comment]
You gueys really want to lose all Chinese users? Stupid choice! Really stupid!

**Comment by jeenlee** [comment]
知乎观光团前来支援

**Comment by Findus23** [comment]
While I don’t want to comment on the topic, I really think that posting the same sentence again and again and splitting the discussion over multiple github issues doesn’t progress the discussion.

**Comment by moshoujingli** [comment]
知乎观光团前来志愿。
https://github.com/contact/report-abuse
各位搜集下法理证据，直接投诉不知道有用没有。
感觉这玩意不跟台湾一样好找。没有啥很打脸的证据啊。

**Comment by huruji** [comment]
Tibet is part of China!

**Comment by qiuai** [comment]
知乎&V2EX观光团前来支援~

**Comment by WuChuanming** [comment]
**Tibet is part of China!**

**Comment by huruji** [comment]
知乎观光团前来支援，You gueys really want to lose all Chinese users? Stupid choice! Really stupid!

**Comment by resouer** [comment]
@sgiehl Here is the PR to fix this: https://github.com/piwik/piwik/pull/6749
But it is closed immediately: https://github.com/piwik/piwik/issues/6994#issuecomment-69875712

**Comment by moshoujingli** [comment]
说起来感觉要不要跟外交部投诉一下http://piwik.org，看看外交手段有啥用处没有

**Comment by sgiehl** [comment]
While till yesterday I tried to have a useful discussion on how this issue could be solved, it seems most people simply want to troll here now. 
So finally this discussion is at some point where I will switch off. I'm doing this project for fun and thus will concentrate on issue being fun for me.
Congrats to all those childish spammers and trolls - I'm out.
Let me know as soon as everyone is ready to have a real discussion on how to solve this issue rather than expressing their political sense (as this is nothing I wanted to discuss) - Maybe I'll consider having a look then again.

**Comment by TesterCC** [comment]
@sgiehl  @mattab We still hope you can change the real-map display issue.  Tibet and Taiwan is a part of China whatever you think.
 
It just a simple modify, we didn't know why you don't change it and always give us all kind of ridiculous explanation since 2014.

If you don't change it, please tell all Chinese programmer directly (maybe you can write it in README), instead of just said "We have a useful discussion on how this issue could be solved" (After all, the earliest related issue reported in 2014)

Please have a basic political baseline though the Internet is free!!!

P.S.
I agree with @xgz123 , if piwik didn't change the real-map display issue, why don't boycott piwik.
Owner has right don't change the map display, users have right don't use it !!!
I think any real Chinese won't use the project which think Tibet and Taiwan didn't a part of China.

What the most important: Tibet and Taiwan is a part of China.

**Comment by phodal** [comment]
@TesterCC had you already see PR in https://github.com/piwik/piwik/pull/11930

Piwik 的开发者们已经做出改变了，见 https://github.com/piwik/piwik/pull/11930

**Comment by TesterCC** [comment]
@phodal  I see the #11930, but piwik demo website still have display the issue. Hope it will be update in the future.

**Comment by phodal** [comment]
@TesterCC it would be, let's keep watching. Because the PR wasn't be merged.

**Comment by mattab** [comment]
Putting Tibet map was a small act of raising awareness about the military invasion of Tibet by China in 1950, and the subsequent destruction of Tibetan culture and freedom of thought by China. China occupation has resulted in the deaths of hundreds of thousands of Tibetans and the imprisonment and torture of thousands more. Plus the destruction of shrines, monasteries...

For most Chinese people, this is actually a good thing as according to them Tibet needed modernization, alphabetisation, etc. and that part was true. But it is really bad how China invaded Tibet and didn't leave these people to sort out their modernization needs on their own, rather than force it upon them. China could have helped Tibet to become a great nation with funding and kindness and respect rather than invade and kill them. And we know that Tibet is a huge natural resource for China (fresh water, mineral). As always, it's not so much about the "great china nation" as it really is about the natural resources.

For reference, similar invasions happened recently in western countries such as the USA when they destroyed the native American culture, or in Australia when they destroyed the aboriginal culture. It could and should have been done differently in an ideal world. But we don't live in this ideal world yet and China did indeed successfully invade Tibet with their army, and killed 1+ million people there.

I think that Chinese people are blinded by the extremely potent propaganda by their leaders, school system, etc. It's interesting how easy it is to influence human beings in believing what you want them to believe, if you control the school system and media. Obviously in our western society we are also blinded by propaganda about consumerism, capitalism, nationalism, and all the rest. It's not just Chinese who are blinded by propaganda.

I like the idea of having put the huge nation of Tibet on the map of Piwik Analytics for the last 4 years. But now that some other team members are pushing for restoring the "standard world map", it is hard for me alone to resist this push by my other colleagues.

**Comment by phodal** [comment]
@mattab I don't agree with you, because a few hundred years ago, the Tibet is a part of China. It means it's not a invade, it's a recapture.  But since the 1900s, a lot of country invade China, they separated the Mongolia, the separated a lot of land from China, and now they want to separate again. We can't bear it, again, and again.

@mattab just another simple question, when we are talking about liber. Why Sikkim is a part of India in Piwik?

**Comment by mattab** [comment]
> It means it's not a invade, it's a recapture.

Yes, China re-captured with military invasion. Same thing. China killed 1+ million Tibetans since 1950, to ensure the re-capture was not reversed.

> Why Sikkim is a part of India in Piwik?

Because a referendum in 1975 led to Sikkim joining India as its 22nd state. Maybe you haven't heard of democracy and referendum as part of your schooling education, but it is an interesting concept where the people (here, Sikkim) chose to be part of India back then.

**Comment by xgz123** [comment]
> Yes, China re-captured with military invasion. Same thing. 
@phodal 我不觉得你能说服他。😂

**Comment by jijiechen** [comment]
@mattab  I'm a little curious how did you know "China killed 1+ million Tibetans", what made you believe so?
These are just your imaginations.

**Comment by desmear** [comment]
It's funny,  these Chinese had use VPN or something else to get through the Great Fire Wall (which is developed by their gov to prevent them) to reach this thread.

**Comment by fengkaijia** [comment]
Let me rephrase @mattab's comment from the Central Tibetan Administration point of view:

> But it is really bad how China invaded Tibet and didn't leave these people to sort out their modernization needs on their own, rather than force it upon them.

We want our slaves back. The communists took away slavery from us.

> China could have helped Tibet to become a great nation with funding and kindness and respect rather than invade and kill them.

Yes, the communists should at first support our system. And make sure we can keep slavery. We attacked them because they cannot promise that.

> As always, it's not so much about the "great china nation" as it really is about the natural resources.

Have you ever read Henry Kissinger's books? You better not read it. It totally makes non-sense. Our theory is better. People, people, people, always remember that China will invade your country when they ran out of resource. Yes, China is dark and dangerous, and please follow Dalai Lama. And make sure you donate to us.

----

The funny thing is that @mattab is citing every bit of information from CTA-controlled or CTA-favored sites without validation. Of source, the best way to validate these claims is go to Tibet and see it by your self, but he didn't. He claims the Chinese are the people blinded by "extremely potent propaganda", in fact, he is. His irreversible bias here let him choose to ignore all contribution the central government made in the past decades. He is blinded to the Qinghai-Tibet Railway or Qinghai-Tibet Highway, and to the fact that Tibet's GDP has increased 27 times from 1993 to 2015 (cite [National Data](http://data.stats.gov.cn/easyquery.htm?cn=E0103))  [Disclaimer: I live in Hong Kong, I don't read Global Times, and I don't need Global Times to tell me about these]. The goal of Western education is to develop a useful citizen for democracy. One who can easily be blind by propaganda does not qualified to that standard.

I thanks "some other team members" and especially, @sgiehl for correcting a four-year long non-sense bug. They made Piwik truely a free software.

And at last, @mattab looks like you knows nothing about the history of China. Go read Henry Kissinger's book.

**Comment by fengkaijia** [comment]
@desmear It's funny that you created your account 30 mins ago and labeled yourself to live in Xi'an. Plus, there's no need to access GitHub via VPN.

**Comment by jijiechen** [comment]
@desmear Do, we do not need VPNs to access GitHub, we get here directly by a direct link.

GFW is another topic which we'd like to discuss with you to prove your understanding is not proper if you have a whole day. Everyone loves the freedom of Internet, but nobody may use it to hurt the freedom of Internet and human being. Think about what happened on Facebook and Twitter before the 2008 Olympic games, during passing of the Olympic torch. That's the real reason why some of the websites are blocked form China.

That's why we think you get so much bad stereotypes on Chinese people and China government.

**Comment by fengkaijia** [comment]
@desmear, and for the GFW issue, read [https://www.quora.com/Why-is-Facebook-banned-in-China-2/answer/Robin-Daverman?srid=3XPCO](https://www.quora.com/Why-is-Facebook-banned-in-China-2/answer/Robin-Daverman?srid=3XPCO).

**Comment by desmear** [comment]
@fengkaijia The chinese gov governs ppl with force and fear, thanks to them, I'd like to be anonymous.
I love Xi'an and China, but not the gov.

**Comment by desmear** [comment]
@fengkaijia @jijiechen Chinese gov blocked most site's overseas including Github, may be Github is unbanned temporarily in some areas, but who knows, they will put it back in the blacklist any time as they wish.
By the way, there's no internet in tibet at all, gov blocked all information from or to tibet.

**Comment by jijiechen** [comment]
@desmear Sham on you. How could you spread such rumours? What you said is not true. Nobody get troubles on access GitHub. Only certain websites are blocked because of political issues on their websites. Tibet has full access to the Internet as other Chinese people do.

**Comment by xgz123** [comment]
这个 @desmear 不知道是个啥东西，散播谣言，懂法的朋友可以动手了

**Comment by JokerQyou** [comment]
Hmm. I thought this issue was to discuss about adding Tibet back to China in the map plugin? I suggest we focus on that instead of talking about GFW. That newly created account is just trolling.

**Comment by desmear** [comment]
@jijiechen How do you know Tibet has full access to the Internet? the gov told you so!
 Can't think by yourself? Sham on you.

**Comment by desmear** [comment]
> 这个 @desmear 不知道是个啥东西，散播谣言，懂法的朋友可以动手了

@xgz123  like I said before, The chinese gov governs ppl with force and fear.
your threatening behaviour proved this.
seems i made the right decision to create this new account via Tor.

**Comment by jijiechen** [comment]
@desmear What a joke! Do people have labels attached on their faces claiming where do they from? I didn't go there in-person, but lots of my friends who went to Tibet for tourism, they published plenty pictures and contents during their vacations online.
It's a waste of my time talking to you.

**Comment by fengkaijia** [comment]
@desmear If there's no Internet in Tibet as you say, then can you explain why someone from Lhasa is visiting my blog hosted in Hong Kong.

![location analytics](https://user-images.githubusercontent.com/3523264/29012617-82f3abf0-7b6f-11e7-8b69-079cbeb27fc4.png)

By the way, that's a screenshot from Google Analytics.

**Comment by xgz123** [comment]
@desmear 直接用中文行么？又要逼我写英语作文。
I think you might have break the law of your country. Is that a threat ? 
I(We) will find you in the flesh -- this is a real threat.
Use new account and tor -- you must be so scared. Not by the goverment, by your self.
I am wasting my time ... 
What you say is not true.

**Comment by mattab** [comment]
@xgz123 Here again you show us who you really are, a strong Chinese patriot threatening your brothers with violence. 

@desmear make sure to report this to Github as threatening of violence is not acceptable in a modern society or on Github and is against the terms and conditions

**Comment by desmear** [comment]
@fengkaijia your picture means your blog have visitors from Tibet , does not mean they're Tibetan.
talking about data, take sina weibo the largest Social Network Site in China for example,
according to an user report by sina weibo I've read months ago, the amount of users from Tibet is almost zero.
you can find and read the report if you like.

**Comment by fengkaijia** [comment]
@desmear You are making claims without citing sources. How on Earth do I know how to find a "report by sina weibo I've read months ago"? If that's your point of persuasive, then I have many Tibetian classmates, you can find them on WeChat.

**Comment by YouYue123** [comment]
@desmear  

As a open-source project contributed by a large community, they should keep their political opinion as neutral as possible. 

Now [UN world map](http://www.un.org/Depts/Cartographic/map/profile/world.pdf) indicates that both of Tibet and Taiwan are part of China clearly. From a neutral point of view, they should  respect the result from the most trustworthy organization in the world

**Comment by moshoujingli** [comment]
> Chinese gov blocked most site's overseas including Github, may be Github is unbanned temporarily in some areas, but who knows, they will put it back in the blacklist any time as they wish.

LOL，How can a adult say such ...
Well, of course, @mattab @desmear you can hold you opinion, and I, maybe some of us, come to here(not across VPN, thanks for the gov) is not to caring about what you are thinking, but to correct this map, in any legal way. If there is a convenient way, that's great. But if not, as I said in the past comment, marking Taiwan  not as a part of the PRC on a public service is against the " UN General Assembly Resolution 2758"(I hope you guys know it.) and other relevant files, I think InnoCraft Ltd. is in the New Zealand ,a menber of UN , so it will follow these policies and ,as a commercial entity, it won't be glad to see this thread become a Diplomatic accident. I'm not sure If you can  acknowledged that the moment the event happening here become bother then bring this to the Ministry of Foreign Affairs or some other UN official  department. And as you said ToS of Github, I think Github is alse a  commercial entity, too many political trouble will let then find out the source.
我个人觉得大家是否考虑给外交部和新西兰大使馆写信投诉piwik的老家，我粘贴出来的那家公司？外交渠道直接怼赞助商和运营者。

**Comment by moshoujingli** [comment]
@desmear
Tibetian, well, they have a nice alias.You call us Chinese, but why not call me Sandonian ? Because I'm not as famous as the killer Da Lai? LoL.

While I come to Youtube and Facebook to see how CNN fool the America, you sham on us by your imagination, Interesting.
And your logical thinking ability seems weak, Can anyone say: While I leaved in Tokyo, I never see a people from Iceland on the internet, because Chinese Gov cant ban me there, so that must be Iceland is blocked from internet by their gov or Japanese banned me from Iceland.

If you so confident with your opinion, just show me the statistic data based on official files or notifications which is acknowledged and will not be challenged by the Permanent members of the United Nations Security Council. I have no interest to hear about  YOU HAVE READ. As referred above, content with these prefix seems nonsense.

**Comment by desmear2** [comment]
OK, OK, I thought keywords are quite easy to google it,
Now, i googled it for you, here's the direct link
http://data.weibo.com/report/reportDetail?id=346
check page 8.

The conversation is over, my previous account was suspended for no reason.
I don't bother keep create new accounts to argue with you for such meaningless topic.
![image](https://user-images.githubusercontent.com/30886160/29153016-7cd86ec2-7dbd-11e7-89bf-b0e5969ddfae.png)

**Comment by fengkaijia** [comment]
@desmear2 Did you seriously read the text? Simply it's gray doesn't mean no one is using it.

READ the text bro: >0.0%. It means few people are using Weibo in Tibet.

There are 3.1 million people live in Tibet, and there are 1.379 billion in total, that means population in Tibet only accounts ~0.25% of the total population in China. The second level in this chart is 0.5%, therefore there's no way the Tibet part can be green unless every Tibetian Weibo user has more than 2 accounts.

The same goes to Qinghai province that only has 5.2 million in population (~0.4%), which is also gray.

![powerpoint](https://user-images.githubusercontent.com/3523264/29157358-2e2b4058-7dd9-11e7-8d3a-d760a164bff2.png)

So where is your critical thinking bro....

**Comment by fengkaijia** [comment]
@desmear2 BTW, your account was suspended because of registering using Tor/not providing your real information. That violates the Account Requirements in TOS.

READ the TOS and before you think it was reported by users.

**Comment by Chriser001** [comment]
@desmear2 Can not understand the contents of the random analysis, that their knowledgeable,【  】是个好东西。But , you did not.

**Comment by zaizai0935** [comment]
都什么时代了，支持藏独和台独的都是傻逼
